src/typing/type_inference_js.ml

```diff
@@ -1135,7 +1133,14 @@ and statement cx = Ast.Statement.(
   | (loc, Empty) -> ()
 
   | (loc, Block { Block.body }) ->
-      toplevels cx body
+      let entries = ref SMap.empty in
+      let block = { scope = Block; entries } in
+      Env_js.push_env block;
+
+      List.iter (statement_decl cx) body;
```

Comment:
Wan to call out this bit specifically. Is it wrong to jump back into the `statement_decl` codepath from the `statement` codepath like this? If this is wrong, then that somewhat invalidates my approach here.